pkg/agent/ssh/server.go

```diff
@@ -101,6 +106,22 @@ func (s *Server) handlePty(session ssh.Session, ptyReq ssh.Pty, winCh <-chan ssh
 	cmd.Env = append(cmd.Env, fmt.Sprintf("TERM=%s", ptyReq.Term))
 	cmd.Env = append(cmd.Env, os.Environ()...)
 	cmd.Env = append(cmd.Env, fmt.Sprintf("SHELL=%s", shell))
+
+	if session.User() != "root" {
+		newHome := "/home/" + session.User()
+
+		// Remove existing HOME variable if it exists
+		for i, v := range cmd.Env {
```

Comment:
This is unnecessary. Latter env variables in the list just take over. You can just append the new HOME.